### DISCUSSION TITLE
Shutter mode "one button" 1 Ch

### DISCUSSION THREAD

**Discussion by sebimt301** [arendst/Tasmota#11625] [Q&A] [answered]
Hello,
Please somebody help me.
I would like to us the shutter mode with 1gang Sonoff TX. So the hutter motor working the following method/sequence:

- Press: up
- Press: stop
- Press: down
- Press: stop

Could you tell me please how can I set it? I tried the Shuttermode3 and otheres but does not work correctly.
Thank you for your help!

**Comment by Pewibe** [comment, marked as answer]
I don't have a dedicated answer for you because of missing expertise. Maybe you read issue #9098 which introduced the mode you want to use. Hopefully something shows you how to go forward. If nothing helps and nobody answers, "stefanbode" is the author of the code and has implemented that "one button mode".

**Reply by sebimt301** [reply]
Could you solve your garage door opener?

**Reply by Pewibe** [reply]
I used the garage door during discussions as an example because I have such a door and I know that thousands of garage doors work like this. Truely I want to use it for a sun shutter on my terrace which works like you asked up-stop-down-stop. I haven't implemented it during winter - nobody used the terrace. Now you trigger me for the upcoming sunshine days that this should be implemented. This is why I wrote "practical exercise is (still) missing".

**Reply by sebimt301** [reply]
I understand. Thank you for your help Pewibe! This would be very important for me, because of tomorrow I have to solve this problem somehow.. :( Do you have any idea or any code how will you try it? I set the ShutterMode 3 which is the garage mode, but it isn't good. And you cannot control with the button so with manually. And unfortunatelly I don't know how could know the shutter which is the actual position and direction. So you know if went from 0% to 50% and I would like to UP to 100% the relay should toggle twice. Because of first go down and stop and after can goes to UP. So it is not easy and now I have no Idea... mhhh :(

**Reply by Pewibe** [reply]
I guess you think more about how the shutter code works ("I don't know how could know the shutter") than how to use it. Using the garage mode means "pressing one button to open-stop-close". The shutter knows the direction only which is reversed by every 2nd press of the controlling button. The percentage of opening the shutter knows by estimating position by time of opening from 0% to 100%. And like a real garage door, if power disappears and returns the door is somewhere unknown between 0% and 100%. Door controls do a door close after power is back to reach the 0% position and set the direction to "open". Thats all. 
If you set "garage mode", press your button, and nothing works or something works wrong, you should write a bug report. 
If you are firm with C-code you can debug your problem in the module "xdrv_27_shutter.ino". stefanbode is the author of that module.

**Reply by sebimt301** [reply]
Maybe successfully done. It seems it is working. The solution is simple. "shuttermode 3" command. I can test it with the shutter only tomorrow, but the console shows good result! :)

So the method is the following:
1. Configure module to the right (for me Sonoff T1CH)
2. SetOption 80 1
3. restart 1
4. shuttermode 3
5. pulsetime1 5

**Reply by sebimt301** [reply]
Thank you for your help Pewibe!!!!

**Reply by Pewibe** [reply]
Fine that you get forward. I looked to the doc "shutter" and remembered something where I'm not sure. The doc is talking about 2 relays even for mode 3. It could be that you have to define a 2nd "virtual" relay beside the first physical one that it works. I tried to convince the author that this looks strange but I do not know what was the end. The doc is still talking about 2 relays.

**Reply by sebimt301** [reply]
Yes you are right, but it seems to working with one relay as well without any virtual relay.
